### ISSUE TITLE
Content showing twice using Inertia SSR with hydrateRoot and laravel-react-i18n

### ISSUE THREAD

**Issue by TheCoati** [EugeneMeles/laravel-react-i18n#16]
When using [laravel-react-i18n](https://github.com/EugeneMeles/laravel-react-i18n) with [Inertia SSR](https://inertiajs.com/server-side-rendering), the page is rendered twice using `hydrateRoot`.

`ssr.tsx`
```ts
void createServer(page => createInertiaApp({
    page,
    render: ReactDOMServer.renderToString,
    resolve: name => {
        const pages = import.meta.glob('./pages/**/*.tsx', { eager: true })
        return pages[`./pages/${name}.tsx`]
    },
    setup: ({App, props}) => {
        return (
            <LaravelReactI18nProvider
                    locale={'en'}
                    fallbackLocale={'en'}
                    files={import.meta.glob('/lang/*.json', { eager: true })}>
                <App {...props} />
            </LaravelReactI18nProvider>
        )
    }
}))
```

`app.tsx`
```ts
void createInertiaApp({
    resolve: name => {
        const pages = import.meta.glob('./pages/**/*.tsx', { eager: true })
        return pages[`./pages/${name}.tsx`]
    },
    setup({ el, App, props }) {
        hydrateRoot(el, <LaravelReactI18nProvider
                locale={'en'}
                fallbackLocale={'en'}
                files={import.meta.glob('/lang/*.json')}>
            <App {...props} />
        </LaravelReactI18nProvider>)
    },
})
```

`home.tsx`
```tsx
export default function HomePage(): ReactElement {
    return (
        <div>
            Test
        </div>
    )
}
```

Restults in:
![image](https://github.com/EugeneMeles/laravel-react-i18n/assets/9860488/d0b6c9f6-a1c1-4202-bf82-25f903edc80b)

---
After removing the `<LaravelReactI18nProvider>`, the text **Test** correctly renders only once, as intended.

**Comment by saullo** [comment]
Any ideias? Same issue here

![image](https://github.com/EugeneMeles/laravel-react-i18n/assets/5266522/0482017c-b51a-4142-95c7-0a04ba441177)

**Comment by saullo** [comment]
I started to have a guess, this is from react.dev docs:

![Screenshot_20240624-231711_Chrome.jpg](https://github.com/EugeneMeles/laravel-react-i18n/assets/5266522/67a76cf0-bc41-4efd-a8a4-fb6b0e8c76a5)

And they explicit say "Using checks like typeof window !== 'undefined' in your rendering logic"

And this file is doing exactly that:

https://github.com/EugeneMeles/laravel-react-i18n/blob/20b330e0e6e41165d7d8e22704449c709aa10c60/src/provider.ts

**Comment by wescopeland** [comment]
I resorted to forking the project to resolve this issue.

The bug is in _provider.ts_. If you want to fix your own project, copy the changes in this file: https://github.com/RetroAchievements/RAWeb/pull/2723/files/b9644c05d1f193032d15f37277b1d88535047e50#diff-f7d458b3d1d0014c6c7114fd9fe84f22424ecd60dca8404f506ad0bc218c6ad7

**Comment by EugeneMeles** [comment]
fixed at 2.0.5
